src/state_point.F90

```diff
@@ -658,7 +658,10 @@ subroutine load_state_point()
     file_id = file_open(path_state_point, 'r', parallel=.true.)
 
     ! Read filetype
-    call read_dataset(file_id, "filetype", int_array(1))
+    call read_dataset(file_id, "filetype", word)
+    if (trim(word) /= 'statepoint') then
```

Comment:
`trim` is not needed for character comparisons. Character variables are space-padded to be of equal length for the purpose of comparison.